Library/Formula/infer.rb

```diff
@@ -0,0 +1,54 @@
+class Infer < Formula
+  desc "Static analyzer for Java, C and Objective-C"
+  homepage "http://fbinfer.com/"
+  url "https://github.com/facebook/infer/releases/download/v0.5.0/infer-osx-v0.5.0.tar.xz"
+  sha256 "6a8547ac0b75a5e2bbeccae2169e39f753a60adbcacb6c94599fd31343a71ce7"
+
+  option "without-clang", "Build without C/Objective-C analyzer"
+  option "without-java", "Build without Java analyzer"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "opam" => :build
+
+  def install
+    if build.without?("clang") && build.without?("java")
+      odie "infer: --without-clang and --without-java are mutually exclusive"
+    end
+    opamroot = buildpath/"build"
+    opamroot.mkpath
+    ENV["OPAMROOT"] = opamroot
+    ENV["OPAMYES"] = "1"
+
+    system "opam", "init", "--no-setup"
+    system "opam", "update"
+
+    system "opam", "install", "ocamlfind"
+    system "opam", "install", "sawja>=1.5.1"
+    system "opam", "install", "atdgen>=1.6.0"
+    system "opam", "install", "extlib>=1.5.4"
+
+    target_platform = if build.without?("clang")
+      "java"
+    elsif build.without?("java")
+      "clang"
+    else
+      "all"
+    end
+    system "./build-infer.sh", target_platform, "--yes"
+
+    if build.with?("clang")
+      libexec.install "facebook-clang-plugins"
+    end
+
+    rm "infer/tests/.inferconfig"
+    libexec.install "infer"
+
+    bin.install_symlink libexec/"infer/bin/infer"
+  end
+
+  test do
+    system "#{bin}/infer", "-v"
```

Comment:
Please bring back the great tests that you added in a previous iteration. And please run them for both languages unconditionally, i.e., without the `if build.with?("clang")` etc. I suggested before my fellow maintainers asked to do otherwise. (The consensus I wasn't aware of is to assume a default installation in `test do`. Sorry for misguiding you. I wasn't aware of that.)